SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp

```diff
@@ -116,3 +136,88 @@ void USpatialDispatcher::ProcessOps(Worker_OpList* OpList)
 		}
 	}
 }
+
+bool USpatialDispatcher::IsExternalSchemaOp(Worker_Op* Op) const
+{
+	Worker_ComponentId ComponentId = GetComponentId(Op);
+	return SpatialConstants::MIN_EXTERNAL_SCHEMA_ID <= ComponentId && ComponentId <= SpatialConstants::MAX_EXTERNAL_SCHEMA_ID;
+}
+
+void USpatialDispatcher::ProcessExternalSchemaOp(Worker_Op* Op)
+{
+	auto TryUserCallback = [&](Worker_ComponentId ComponentId, TFunction<void(UOpCallbackTemplate*)>& OpCallback)
+	{
+		if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(ComponentId))
+		{
+			UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+			OpCallback(UserCallback);
+		}
+	};
+	Worker_ComponentId ComponentId = GetComponentId(Op);
```

Comment:
Since we check `IsExternalSchemaOp` before calling `ProcessExternalSchemaOp`, we should already know we're dealing with one of the ops we care about (otherwise `GetComponentId` would have returned `INVALID_COMPONENT_ID` and `IsExternalSchemaOp` would be false).
Add a sanity `check`, and you don't need the roundabout lambdas.
```
Worker_ComponentId ComponentId = GetComponentId(Op);
check(ComponentId != SpatialConstants::INVALID_COMPONENT_ID);

UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(ComponentId);
if (UserCallbackWrapper == nullptr)
{
	// Log that there's unhandled external schema
	return;
}
UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
switch (Op->op_type)
{
case WORKER_OP_TYPE_ADD_COMPONENT:
	UserCallback->OnAddComponent(Op->add_component);
// other ops...
default:
	// GetComponentId should have only returned a valid ID for the ops above
	checkNoEntry();
}
```